lib/iris/fileformats/pp.py

```diff
@@ -1297,6 +1301,16 @@ def core_data(self):
         return self._data
 
     @property
+    def data_dtype(self):
+        return self._data.dtype \
+            if self._data_dtype is None \
+            else self._data_dtype
```

Comment:
Well, I dunno if it would improve it. You'd still have the confusion because of the similarly named variables. Maybe I should just rename `_data_dtype` to something else. In fact, now that I think about it, the name might be confusing anyway, because the data already _has_ a dtype, but it's not the one we want. Maybe I should call it `realised_dtype`, like in `DataManager`.